### PR TITLE
Migrate VM Bug Fix

### PR DESCRIPTION
**Pull request by naveena-maplelabs**
Issue
1) Duplicate resource pool in the vcenter caused unexpected error. 2) Raise error incase of missing resources and mark required field.

Fix:
1) Added option cluster_compute_resource to filter duplicate resource pools.
2) List of mandatory fields are added as required parameter. 3) Improved checkmode code.

### CODE REVIEW

File: plugins/modules/cohesity_migrate_vm.py [hunk 1]
```diff
@@ -560,15 +607,16 @@ def main():
             endpoint=dict(type="str", required=True),
             environment=dict(choices=["VMware"], default="VMware"),
             job_vm_pair=dict(type="dict", required=True),
-            datastore_name=dict(type="str", default=""),
+            datastore_name=dict(type="str", required=True),
```
Comment (pyashish) [inline plugins/modules/cohesity_migrate_vm.py:610]: Datastore name is required ? This breaks compatibility ?

**Review by pyashish — CHANGES_REQUESTED** [review]
Does check mode for each resource work ?

File: plugins/modules/cohesity_migrate_vm.py (hunk 1 above)
Comment (naveena-maplelabs) [reply plugins/modules/cohesity_migrate_vm.py:610]: Resource pool and datastore name are required parameters.

Comment (pyashish) [reply plugins/modules/cohesity_migrate_vm.py:680]: But Datastore isn't required in UI correct ?

**Pull request merged**